mfr/extensions/tabular/libs/stdlib_tools.py

```diff
@@ -30,3 +34,21 @@ def csv_stdlib(fp):
         })
     rows = [row for row in reader]
     return columns, rows
+
+
+def detect_quoted_fields(dialect, data):
```

Comment:
Regarding `re.compile`'ing, that would only yield a performance benefit if it was done in the global scope. Compiling four regexes each time this function is called would, at best, yield no change in performance. I'd also argue that this function isn't called frequently enough to merit global regex compiling